src/api.rs

```diff
@@ -1529,11 +1529,21 @@ impl<T: Pixel> ContextInner<T> {
 
               // Coordinates of the top-left corner of the block intersecting the
               // reference block from the top-left.
-              let top_left_block_x =
+              let mut top_left_block_x =
```

Comment:
Something along the lines of:
``` rust
let top_left_block_x = (reference_x - if (reference_x < 0) { MI_SIZE_IN_MV_UNITS - 1 } else { 0 }) / MI_SIZE_IN_MV_UNITS * MI_SIZE_IN_MV_UNITS
```
Wouldn't be equivalent?

Reply:
It should be I think.

Reply:
Please update that way if it works :)